src/tests/unit/tests/DetailsView/components/adhoc-tab-stops-test-view.test.tsx

```diff
@@ -98,8 +102,18 @@ describe('AdhocTabStopsTestView', () => {
                 isChanged: true,
             };
 
+            props.deps = {
+                detailsViewActionMessageCreator: detailsViewActionMessageCreatorMock.object,
+            } as AdhocTabStopsTestViewDeps;
+
+            detailsViewActionMessageCreatorMock
+                .setup(m => m.targetPageChangedResetData())
+                .verifiable();
+
             const wrapper = shallow(<AdhocTabStopsTestView {...props} />);
             expect(wrapper.getElement()).toMatchSnapshot();
+
+            detailsViewActionMessageCreatorMock.verifyAll();
```

Comment:
Could you put the action message verification in a separate test case from the rendering snapshot test, so we can avoid having to include a snapshot of a mock proxy object? Proxy snapshots tend to require extraneous snapshot updates when typemoq/swc/typescript/node versions update.

Reply:
With my latest update this is no longer an issue, but I also don't fully follow how to do this? The mock verification doesn't change the snapshot as far as I know, but the mock did need to exist in the props for the view to render, because `props.deps.detailsViewActionMessageCreator.targetPageChangedResetData()` was being called. How do we get around having a proxy object in the snapshot in a scenario like that?

Reply:
The two options I use most often are:

* If the snapshot test in question doesn't need to capture a "deep" copy of the props to be acting as a valuable test, you can snapshot `theEnzymeShallowInstance.debug()` instead of `theEnzymeShallowInstance.getElement()`; this produces a "shallower" snapshot that doesn't encode so much implementation detail of the mock proxy
* If the test is more valuable for verifying other "deep" subproperties of props, you can replace the use of `Mock.ofInstance`/etc with a less-smart stub object. Usually this route is much more realistic if you can separate out the tests of the rendering (which can use simple stub objects) vs tests of callback behavior (which are easier to write with full mock objects). For example, here, you might have the rendering test pass a `detailsViewActionMessageCreator` whose value was `{ targetPageChangedResetData: () => {} }`, rather than `Mock.ofType<DetailsViewActionMessageCreator>()`

Reply:
This is great, thanks for taking the time to explain it!